component/core/src/test/java/org/exoplatform/social/core/profile/ProfilePropertySettingsServiceTest.java

```diff
@@ -0,0 +1,158 @@
+    assertEquals("Profile Property Setting Technical Identifier is mandatory", exception1.getMessage());
+    try {
+      ProfilePropertySetting profilePropertySetting = createProfileSettingInstance("test");
+      profilePropertySettingsService.createPropertySetting(profilePropertySetting);
+      profilePropertySetting = profilePropertySettingsService.getProfileSettingByName(profilePropertySetting.getPropertyName());
+      assertNotNull(profilePropertySetting.getId());
+      profilePropertySettingsService.deleteProfilePropertySetting(profilePropertySetting.getId());
+      profilePropertySetting = profilePropertySettingsService.getProfileSettingByName(profilePropertySetting.getPropertyName());
+      assertNull(profilePropertySetting);
+    } catch (Exception e) {
+      fail(e.getMessage());
+    }
+  }
+
+  public void testUpdateProfilePropertySetting() {
+    try {
+      ProfilePropertySetting profilePropertySetting = createProfileSettingInstance("test");
+      profilePropertySettingsService.createPropertySetting(profilePropertySetting);
+      profilePropertySetting = profilePropertySettingsService.getProfileSettingByName(profilePropertySetting.getPropertyName());
+      assertTrue(profilePropertySetting.isActive());
+      profilePropertySetting.setActive(false);
+      profilePropertySettingsService.updatePropertySetting(profilePropertySetting);
+      profilePropertySetting = profilePropertySettingsService.getProfileSettingByName(profilePropertySetting.getPropertyName());
+      assertFalse(profilePropertySetting.isActive());
+    } catch (Exception e) {
+      fail(e.getMessage());
+    }
+  }
+
+  public void testGetProfilePropertySettings() {
+    try {
+      assertEquals(0, profilePropertySettingsService.getPropertySettings().size());
+      ProfilePropertySetting profilePropertySetting1 = createProfileSettingInstance("test10");
+      ProfilePropertySetting profilePropertySetting2 = createProfileSettingInstance("test20");
+      ProfilePropertySetting profilePropertySetting3 = createProfileSettingInstance("test30");
+      ProfilePropertySetting profilePropertySetting4 = createProfileSettingInstance("test40");
+      ProfilePropertySetting profilePropertySetting5 = createProfileSettingInstance("test50");
+      ProfilePropertySetting profilePropertySetting6 = createProfileSettingInstance("test60");
+      profilePropertySettingsService.createPropertySetting(profilePropertySetting1);
+      profilePropertySettingsService.createPropertySetting(profilePropertySetting2);
+      profilePropertySettingsService.createPropertySetting(profilePropertySetting3);
+      profilePropertySettingsService.createPropertySetting(profilePropertySetting4);
+      profilePropertySettingsService.createPropertySetting(profilePropertySetting5);
+      profilePropertySettingsService.createPropertySetting(profilePropertySetting6);
+      assertEquals(6, profilePropertySettingsService.getPropertySettings().size());
```

Comment:
please test on order of retrieved properties

Reply:
Done